gum/backend-darwin/gumprocess-darwin.c

```diff
@@ -1380,6 +1478,50 @@ gum_darwin_enumerate_ranges (mach_port_t task,
   }
 }
 
+static gboolean
+gum_darwin_fill_file_mapping (gint pid,
+                              mach_vm_address_t address,
+                              GumFileMapping * file,
+                              struct proc_regionwithpathinfo * region)
+{
+  gint retval, len;
+
+  retval = __proc_info (PROC_INFO_CALL_PIDINFO, pid, PROC_PIDREGIONPATHINFO,
+      (uint64_t) address,  region, sizeof (struct proc_regionwithpathinfo));
+
+  if (retval == -1)
+    return FALSE;
+
+  len = strnlen (region->prp_vip.vip_path, MAXPATHLEN);
+  region->prp_vip.vip_path[len] = '\0';
+
+  if (len == 0)
+    return FALSE;
+
+  file->path = region->prp_vip.vip_path;
+  file->offset = region->prp_prinfo.pri_offset;
+  file->size = region->prp_vip.vip_vi.vi_stat.vst_size;
+
+  return TRUE;
+}
+
+static void
+gum_darwin_adjust_range_size (GumMemoryRange * range,
```

Comment:
How about naming this `gum_darwin_clamp_range_size()` so it's clearer what kind of adjustment is being made?